gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/DatasetsFinder.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.compaction.mapreduce;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Maps;
+
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.compaction.Dataset;
+import gobblin.configuration.State;
+import gobblin.util.DatasetFilterUtils;
+import gobblin.util.HadoopUtils;
+
+
+/**
+ * {@link Dataset}s finder to identify datasets, using given properties.
+ */
+@Slf4j
+public abstract class DatasetsFinder {
```

Comment:
This should implement gobblin.data.management.retention.dataset.finder.DatasetFinder<T>